### DISCUSSION TITLE
Using the pimcorecache tag with a dynamic key

### DISCUSSION THREAD

**Discussion by muratbinerbay** [pimcore/pimcore#17870] [Q&A / Support]
Hi everyone,

We used to init the old `pimcore_cache` extension like:

```
    {% set cache = pimcore_cache("product_detail_" ~ product.id) %}
``` 
Which allowed us to create specific cache keys based on the product IDs.

With the new `pimcorecache` tags it looks like the first token **must** be of type string and cannot be a Twig variable. It looks like the `tags` argument behaves similarly.

We tried a few things to some how get a string value passed into the CacheParser but it was not possible. Is this intentional? Wouldn't allowing dynamic cache keys be helpful in some cases?

For reference, this change was added in #17444 

Thank you,
Murat

**Comment by muratbinerbay** [comment]
/cc @markus-moser @kingjia90 
wanted to ping you and ask again if the change is intentional before I try to make changes & submit a PR :)

Thanks!

**Comment by brusch** [comment]
@muratbinerbay I agree that using dynamic cache keys would be beneficial. Imho this change was not intentional as there are no arguments against it. 

A PR would be very nice, thanks! 👍

**Reply by hadl** [reply]
Sorry, i have to ask a stupid question:
What is the difference between the native twig `cache` and `pimcorecache` tags?
Do we really need `pimcorecache`?

**Reply by markus-moser** [reply]
Yes we need `pimcorecache` but it's built the same way like `cache`. The only real difference is that `pimcorecache` also supports the `force` parameter to either force caching in editmode or to show the uncached version. The regular `cache` does not know that you are in the editmode and therefore always force it.

**Reply by hadl** [reply]
Thx @markus-moser
